lib/delighted/http_response.rb

```diff
@@ -37,5 +42,17 @@ def get_header_value(key)
         end
       end
     end
+
+    def parse_link_header(header_value)
+      links = {}
+      # Parse each part into a named link
+      header_value.split(',').each do |part, index|
```

Comment:
If we intend to allow for the possibility of multiple links, it's worth noting that if multiple Link headers exist in the response (as opposed to a single Link header with multiple links), then `get_header_value` will return an Array rather than a String. So if we want to normalize all of that, we probably need something like:

```
parts = Array(header_value)
  .map { |v| v.split(",") }
  .flatten
  .map(&:strip)
```
It's a bit of a mouthful, but one nice side effect of this code is that it handles `nil` and `""` values for the header value cleanly (they effectively become empty lists) and you won't need that `if !header_value.nil? && !header_value.empty?` conditional on end of the block.

Also, `Array#each` doesn't pass a second block argument, so `index` will always be `nil`; it should just be `parts.each do |part|` (you aren't using that argument anyway).

Reply:
Nice one, will definitely add that.